documentation/technical_writing_slides.md

```diff
@@ -0,0 +1,452 @@
+
+> The best function names in the world aren't going to answer the questions “Why does the software do this?” and “Why doesn’t it do this in a simpler way?”
+
+---
+
+## Be Clear About Your Audience (2/2)
+
+[py-RSE](https://merely-useful.tech/py-rse/documentation.html) distinguishes novices, competent practitioners, and experts (following [Wilson 2019, Teaching Tech Together](https://doi.org/10.1201/9780429330704))
+
+- A **novice** doesn’t yet have a mental model of the domain. -> Needs **tutorials** that introduce key ideas one by one.
+    - Good example: [deal.ii tutorials](https://dealii.org/developer/doxygen/deal.II/Tutorial.html)).
+- A **competent practitioner** knows enough to accomplish routine tasks with routine effort. -> Needs **reference guides, cookbooks, and Q&A sites**.
+    - Good example: [preCICE config reference](https://precice.org/configuration-xml-reference.html)).
+- An **expert** need this material as well – nobody's memory is perfect.
+
+---
+
+## Style Guides (1/4)
+
+Consistent tone and style ...
+
+- makes content easier to read.
+- reduces the reader's cognitive load.
+- increases confidence in authority of content.
+
+Read more: [Write the Docs on style guides](https://www.writethedocs.org/guide/writing/style-guides/)
+
+---
+
+## Style Guides (2/4)
+
+- Different people write differently. Even the same person writes differently every day.
+- Style guide contains a set of standards.
+- Writer does not have to worry -> saves time.
+- Traditional style guides have been around as long as publishing, e.g. [The Chicago Manual of Style](https://www.chicagomanualofstyle.org/).
+
+---
+
+## Style Guides (3/4)
+
+Example: brief excerpt of [Google developer documentation style guide](https://developers.google.com/style)
+
+- **Timeless documentation**: Avoid words such as *"currently"*, *"in the future"*, *"soon"*
+- Use present tense, avoid *"will"*
+    - Bad: *Send a query to the service. The server will send an acknowledgment.*
+
+- **Exclamation points**: Don't use them. Rather use notices such as Note or Caution.
+
+---
+
+## Style Guides (4/4)
+
+- **Headings**: Use "Sentence case of Headings", not "Title Case for Headings"
+- **Image URLs**: Use site-root-relative URL from same domain
+- **Referring to filenames**: code font, word "file" after filename, exact spelling
+    - Good: *In the following `build.sh` file, modify the default values for all parameters.*
+
+- ... even [example person names](https://developers.google.com/style/examples#example-person-names)
+
+---
+
+## Tone of Voice (1/2)
+
+... helps a lot to make tone of documentation consistent.
+
+- If your software/product was a person, how would they be?
+- Let it talk like this person.
+
+> Example preCICE:
+> - Three things to define character? Smart, approachable, responsible.
+> - What kind of smartphone does preCICE use? No iPhone.
+> - Does preCICE have a sense of humor? Yes. When does it use it? Only rarely.
```

Comment:
```suggestion
> - Does preCICE have a sense of humor? Yes. When do they use it? Only rarely.
```
Shouldn't one use "they" as mentioned in the [Google style guide](https://developers.google.com/style/pronouns#gender-neutral-pronouns)?